File: src/Application Specific/Classes/Entity Collections/Tablet Orders/TabletOrderItem.vb
Imports System.Data.SqlClient

Namespace Entity

    Namespace TabletOrders

        Public Class TabletOrderItem

            Private _dataTypeValidator As DataTypeValidator
            Public Sub New()
                _dataTypeValidator = New DataTypeValidator
            End Sub

#Region "Class Properties"

            Public Property IgnoreExceptionsOnSetMethods() As Boolean
                Get
                    Return Me._dataTypeValidator.IgnoreExceptionsOnSetMethods
                End Get
                Set(ByVal Value As Boolean)
                    Me._dataTypeValidator.IgnoreExceptionsOnSetMethods = Value
                End Set
            End Property

            Public ReadOnly Property Errors() As Hashtable
                Get
                    Return _dataTypeValidator.Errors
                End Get
            End Property

            Public ReadOnly Property DTValidator() As DataTypeValidator
                Get
                    Return _dataTypeValidator
                End Get
            End Property

            Private _exists As Boolean = False
            Public Property Exists() As Boolean
                Get
                    Return _exists
                End Get
                Set(ByVal Value As Boolean)
                    _exists = Value
                End Set
            End Property

            Private _deleted As Boolean = False
            Public ReadOnly Property Deleted() As Boolean
                Get
                    Return _deleted
                End Get
            End Property
            Public WriteOnly Property SetDeleted() As Boolean
                Set(ByVal Value As Boolean)
                    _deleted = Value
                End Set
            End Property

#End Region

#Region "Tablet Order Properties"


            Private _TabletOrderID As Integer = 0
            Public ReadOnly Property TabletOrderID() As Integer
                Get
                    Return _TabletOrderID
                End Get
            End Property
            Public WriteOnly Property SetTabletOrderID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_TabletOrderID", Value)
                End Set
            End Property

            Private _EngineerID As Integer
            Public ReadOnly Property EngineerID() As Integer
                Get
                    Return _EngineerID
                End Get
            End Property
            Public WriteOnly Property SetEngineerID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_EngineerID", Value)
                End Set
            End Property

            Private _SupplierID As Integer
            Public ReadOnly Property SupplierID() As Integer
                Get
                    Return _SupplierID
                End Get
            End Property
            Public WriteOnly Property SetSupplierID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SupplierID", Value)
                End Set
            End Property

            Private _SelectedVisitID As Integer
            Public ReadOnly Property SelectedVisitID() As String
                Get
                    Return _SelectedVisitID
                End Get
            End Property
            Public WriteOnly Property SetSelectedVisitID() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_SelectedVisitID", Value)
                End Set
            End Property

            Private _ForNextVisit As Boolean
            Public ReadOnly Property ForNextVisit() As Boolean
                Get
                    Return _ForNextVisit
                End Get
            End Property
            Public WriteOnly Property SetForNextVisit() As Object
                Set(ByVal Value As Object)
                    _ForNextVisit = Value
                End Set
            End Property

            Private _Status As String = String.Empty
            Public ReadOnly Property Status() As String
                Get
                    Return _Status
                End Get
            End Property
            Public WriteOnly Property SetStatus() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_Status", Value)
                End Set
            End Property

            Private _OrderCreated As Date
            Public ReadOnly Property OrderCreated() As Date
                Get
                    Return _OrderCreated
                End Get
            End Property
            Public WriteOnly Property SetOrderCreated() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_OrderCreated", Value)
                End Set
            End Property

            Private _LastUpdated As Date
            Public ReadOnly Property LastUpdated() As Date
                Get
                    Return _LastUpdated
                End Get
            End Property
            Public WriteOnly Property SetLastUpdated() As Object
                Set(ByVal Value As Object)
                    _dataTypeValidator.SetValue(Me, "_LastUpdated", Value)
                End Set
            End Property

#End Region

        End Class

    End Namespace

End Namespace

File: src/Application Specific/Classes/Entity Collections/Tablet Ordering/TabletOrderItem.cs
using System;
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace TabletOrders
    {
        public class TabletOrderItem
        {
            private DataTypeValidator _dataTypeValidator;

            public TabletOrderItem()
            {
                _dataTypeValidator = new DataTypeValidator();
            }

            
            public bool IgnoreExceptionsOnSetMethods
            {
                get
                {
                    return _dataTypeValidator.IgnoreExceptionsOnSetMethods;
                }

                set
                {
                    _dataTypeValidator.IgnoreExceptionsOnSetMethods = value;
                }
            }

            public Hashtable Errors
            {
                get
                {
                    return _dataTypeValidator.Errors;
                }
            }

            public DataTypeValidator DTValidator
            {
                get
                {
                    return _dataTypeValidator;
                }
            }

            private bool _exists = false;

            public bool Exists
            {
                get
                {
                    return _exists;
                }

                set
                {
                    _exists = value;
                }
            }

            private bool _deleted = false;

            public bool Deleted
            {
                get
                {
                    return _deleted;
                }
            }

            public bool SetDeleted
            {
                set
                {
                    _deleted = value;
                }
            }

            
            

            private int _TabletOrderID = 0;

            public int TabletOrderID
            {
                get
                {
                    return _TabletOrderID;
                }
            }

            public object SetTabletOrderID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_TabletOrderID", value);
                }
            }

            private int _EngineerID;

            public int EngineerID
            {
                get
                {
                    return _EngineerID;
                }
            }

            public object SetEngineerID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_EngineerID", value);
                }
            }

            private int _SupplierID;

            public int SupplierID
            {
                get
                {
                    return _SupplierID;
                }
            }

            public object SetSupplierID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SupplierID", value);
                }
            }

            private int _SelectedVisitID;

            public string SelectedVisitID
            {
                get
                {
                    return _SelectedVisitID.ToString();
                }
            }

            public object SetSelectedVisitID
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_SelectedVisitID", value);
                }
            }

            private bool _ForNextVisit;

            public bool ForNextVisit
            {
                get
                {
                    return _ForNextVisit;
                }
            }

            public object SetForNextVisit
            {
                set
                {
                    _ForNextVisit = Conversions.ToBoolean(value);
                }
            }

            private string _Status = string.Empty;

            public string Status
            {
                get
                {
                    return _Status;
                }
            }

            public object SetStatus
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_Status", value);
                }
            }

            private DateTime _OrderCreated;

            public DateTime OrderCreated
            {
                get
                {
                    return _OrderCreated;
                }
            }

            public object SetOrderCreated
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_OrderCreated", value);
                }
            }

            private DateTime _LastUpdated;

            public DateTime LastUpdated
            {
                get
                {
                    return _LastUpdated;
                }
            }

            public object SetLastUpdated
            {
                set
                {
                    _dataTypeValidator.SetValue(this, "_LastUpdated", value);
                }
            }

            
        }
    }
}